File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_map = {}
        for num in nums:
            if num in num_map:
                num_map[num] += 1
            else:
                num_map[num] = 1

        sorted_nums = sorted(num_map, key=num_map.get, reverse=True)
        return sorted_nums[:k]

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int [] res = new int [nums.length];
         res[0]=1;

        for(int i=1;i<nums.length;i++)
        {
            res[i]=res[i-1]*nums[i-1];

        }
        int suffix=1;
        for(int j=nums.length-1;j>=0;j--)
        {
            res[j]*=suffix;
            suffix*=nums[j];





        }
        return res;

        
        
    }
}  
